STM32 HW-test die/digital tempearture Sensor in CI

djiatsaf-st opened the pull request:
- Add overlay files for some specifics boards to be able to run on HW with die_temp_polling sample

- Update sample.yaml to integrate new boards in CI

MaureenHelm reviewed (changes requested):
@avisconti @erwango please take a look

djiatsaf-st commented on samples/sensor/die_temp_polling/sample.yaml:
```diff
@@ -7,8 +7,26 @@ tests:
+    - nucleo_wb55rg
+    - nucleo_wl55jc
+    - stm32f3_disco
+    - stm32h573i_dk
```
Thanks, good point, I'll use platform_allow instead.

djiatsaf-st commented on samples/sensor/die_temp_polling/boards/b_u585i_iot02a.overlay:
```diff
@@ -0,0 +1,22 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/ {
+	aliases {
+		die-temp0 = &die_temp;
+	};
+};
+
+&die_temp {
+	status = "okay";
+};
+
+&adc1 {
+	pinctrl-0 = <&adc1_in5_pa0>;
+	pinctrl-names = "default";
+	st,adc-prescaler = <4>;
+	status = "okay";
+};
```
Not really, I'll put it in the devicetree

djiatsaf-st committed:
boards: st:  add die_temp or digi node

update die_temp node for some stm32 boards,
need that for run sensor die_temp_polling samples.
don't need to modify sample.yaml

Signed-off-by: Fabrice DJIATSA <fabrice.djiatsa-ext@st.com>

gautierg-st commented on boards/st/nucleo_h743zi/nucleo_h743zi.dts:
```diff
@@ -160,7 +160,7 @@ zephyr_udc0: &usbotg_fs {
 
 &adc3 {
-	pinctrl-0 = <&adc3_inp5_pf3>;
+	pinctrl-0 = <&adc3_inp5_pf3 &adc3_inp0_pc2_c>;
```
PC2_C is a special pin that may require a call to the LL_SYSCFG_OpenAnalogSwitch function if PC2 come to be used for something else (to separate the 2 pads). Is there a need to add this specific pin?

gautierg-st commented on boards/st/nucleo_wba55cg/nucleo_wba55cg.dts:
```diff
@@ -62,6 +62,18 @@
 		};
 	};
 
+	die_temp: dietemp {
+
+		compatible = "st,stm32-temp-cal";
+		ts-cal1-addr = <0x0BF90710>;
+		ts-cal2-addr = <0x0BF90742>;
+		ts-cal1-temp = <30>;
+		ts-cal2-temp = <130>;
+		ts-cal-vrefanalog = <3000>;
+		io-channels = <&adc4 13>;
+		status = "okay";
+	};
+
```
This should be in the .dtsi file (except that the status should be `"disabled"` in it).
You should only have `status = "okay";` here.

gautierg-st commented on boards/st/stm32h573i_dk/stm32h573i_dk.dts:
```diff
@@ -53,11 +53,22 @@
 		};
 	};
 
+	soc {
+		digi_die_temp: digi_dietemp@40008C00 {
+			compatible = "st,stm32-digi-temp";
+			reg = <0x40008C00 0x400>;
+			interrupts = <113 0>;
+			interrupt-names = "digi";
+			clocks = <&rcc STM32_CLOCK_BUS_APB1 0x00000008>;
+			status = "okay";
+		};
+	};
+
```
Same remark as for the WBA. Should be in dtsi.

djiatsaf-st committed:
dts: arm: st:  add digi-temp node in dtsi file

add digital temperature node in board.dtsi file

Signed-off-by: Fabrice DJIATSA <fabrice.djiatsa-ext@st.com>

gautierg-st commented on dts/arm/st/wba/stm32wba55.dtsi:
```diff
@@ -9,4 +9,16 @@
 		compatible = "st,stm32wba55", "st,stm32wba", "simple-bus";
 	};
+
+	die_temp: dietemp {
+
+		compatible = "st,stm32-temp-cal";
+		ts-cal1-addr = <0x0BF90710>;
+		ts-cal2-addr = <0x0BF90742>;
+		ts-cal1-temp = <30>;
+		ts-cal2-temp = <130>;
+		ts-cal-vrefanalog = <3000>;
+		io-channels = <&adc4 13>;
+		status = "disabled";
+	};
```
This should also be added in stm32wba.dtsi.
Nit: remove the blank line.

gautierg-st commented on boards/st/disco_l475_iot1/disco_l475_iot1.dts:
```diff
@@ -295,7 +296,7 @@ zephyr_udc0: &usbotg_fs {
 &adc1 {
 	pinctrl-0 = <&adc1_in3_pc2 &adc1_in4_pc3
-		     &adc1_in13_pc4 &adc1_in14_pc5>;
+		     &adc1_in13_pc4 &adc1_in14_pc5 >;
```
Nit: remove the space before >

gautierg-st reviewed (changes requested):
Still a few comments :)

djiatsaf-st commented on boards/st/stm32h573i_dk/stm32h573i_dk.dts:
```diff
@@ -57,7 +57,7 @@
 		sw0 = &user_button;
 		watchdog0 = &iwdg;
-		die-temp0 = &die_temp;
+		die-temp0 = &digi_die_temp;
```
Good point , i will use this configuration instead.  

```
die-temp0 = &digi_die_temp; 
die-temp1 = &die_temp;
```

The sample.yaml have a filter on the aliase "die-temp0" , and the purpose of this test is about digital-temp first .

will this good for you ?

gautierg-st replied:
It doesn't change anything, both sensors will be tested by the sample. The filter looks for the first (index 0), but the code will test all that follow.

MaureenHelm commented on dts/arm/st/h5/stm32h5.dtsi:
```diff
@@ -595,7 +595,7 @@
 			reg = <0x40008c00 0x400>;
 			interrupts = <113 0>;
-			interrupt-names = "digi";
+			interrupt-names = "digi_temp";
```
This change belongs in commit `dts: arm: st: add digi-temp node in dtsi file`

MaureenHelm commented on samples/sensor/die_temp_polling/boards/nucleo_u575zi_q.overlay:
```diff
@@ -0,0 +1,16 @@
+ */
+
+
+/ {
+	aliases {
+		die-temp0 = &die_temp;
+	};
+};
+
+&die_temp {
+	status = "okay";
+};
```
These changes can be moved to the board's dts, don't need an overlay.

MaureenHelm commented on samples/sensor/die_temp_polling/boards/stm32h750b_dk.overlay:
```diff
@@ -0,0 +1,24 @@
+ */
+
+/ {
+	aliases {
+		die-temp0 = &die_temp;
+	};
+};
+
+&die_temp {
+	status = "okay";
+};
+
+&adc3 {
+	pinctrl-0 = <&adc3_inn3_pf8>;
+	pinctrl-names = "default";
+
+	st,adc-clock-source = <SYNC>;
+	st,adc-prescaler = <4>;
+	status = "okay";
+};
```
These changes can be moved to the board's dts, don't need an overlay.
